threading/includes/network_handler.h

```diff
@@ -0,0 +1,90 @@
+#ifndef __NETWORK_HANDLER_H__
+#define __NETWORK_HANDLER_H__
+
+#define _XOPEN_SOURCE 700
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <getopt.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <pthread.h>
+#include <string.h>
+#include <errno.h>
+
+#include "global_data.h"
+
+#define MAX_PORT_LEN 6
+#define BASE_10      10
+#define MIN_PORT     1025
+#define MAX_PORT     65535
+
+/**
+ * @brief - a struct to store server specific data, the port received from
+ *          the command line upon server file execution and the total number
+ *          of allowable connections (either passed at cmdline or specified)
+ *          upon execution
+ * @member port - a string representation of the port passed by the user to setup
+ *                the TCP client connections
+ * @member num_allowable_clients - received either from the command line or within
+ *                                 setup 
+ */
+typedef struct setup_info
+{
+    char   * port;
+    size_t   num_allowable_clients;
```

Comment:
Why is `num_allowable_clients` of type `size_t`? It should be something like `uint16_t` or `uint8_t`. Don't use a type that is implementation dependent for something that should be a predictable, portable range of allowed values. In this case, you'd likely want to enforce some reasonable maximum value (like MAX_UINT8 or MAX_UINT16).